### PR TITLE
hdrgen.d: remove globals.d import

### PR DESCRIPTION
**Pull request by WalterBright**
dinteger_t is a relic from the old C code base, and is no longer relevant because D has predictable sizes of integer types.

### CODE REVIEW

File: compiler/src/dmd/hdrgen.d [hunk 1]
```diff
@@ -2161,11 +2160,8 @@ private void expressionPrettyPrint(Expression e, ref OutBuffer buf, ref HdrGenSt
             default:
                 /* This can happen if errors, such as
                  * the type is painted on like in fromConstInitializer().
+                 * Just ignore
                  */
-                if (!global.errors)
-                {
-                    assert(0);
-                }
```
Comment (thewilsonator) [inline compiler/src/dmd/hdrgen.d:2168]: this looks to be unrelated, please lave for a separate pr

Comment (WalterBright) [reply compiler/src/dmd/hdrgen.d:2168]: It is related, as it requires the import.

**Review by thewilsonator — APPROVED** [review]

**Comment by dkorpel** [comment]
> dinteger_t is a relic from the old C code base, and is no longer relevant

It is still relevant because it represents the compiler's internal integer type, which shouldn't be hard-coded to 64-bits: https://github.com/dlang/dmd/pull/6743#issuecomment-1378683424

Please refrain from replacing it with `long` / `ulong`, though you can replace usages with type inference:

```D
const dinteger_t x = f();
const x = f();
```

**Comment by ibuclaw** [comment]
Yeah, I'm just going to revert all of this back.
